### ISSUE TITLE
Replace deprecated torch.range in rscd/losses/loss_util/criterion_mp.py with torch.arange

### ISSUE THREAD

**Issue by xyf5432** [xwmaxwma/rschange#28]
### Summary

`rscd/losses/loss_util/criterion_mp.py:268` in `SetCriterion_mp.forward` (the DN / denoising training branch) calls `torch.range`, which is deprecated and emits a `UserWarning` on every call ("torch.range is deprecated and will be removed in a future release because its behavior is inconsistent with Python's range builtin. Instead, use torch.arange, which produces values in [start, end)." — verified on torch 2.11.0). It is the only `torch.range` usage in this repo:

```python
for i in range(len(targets)):
    if len(targets[i]['labels']) > 0:
        t = torch.range(0, len(targets[i]['labels']) - 1).long().cuda()
```

Because it sits inside the per-target loop of the forward pass, **every training step in DN mode triggers the warning**, for every target with labels — a warning on every batch during training.

### Proposed fix

`torch.range` is **end-inclusive** — `torch.range(0, L - 1)` produces exactly `L` values (0..L-1). The `torch.arange` equivalent is the same length with the same values:

```python
t = torch.arange(len(targets[i]['labels'])).long().cuda()
```

(`torch.arange` with integer inputs returns `int64`, so the explicit `.long()` is a no-op — kept only to match the surrounding style. Note line 271's `range(scalar)` is the Python built-in, not `torch.range`, and needs no change.)

### Validation (torch 2.11.0+cpu)

- `torch.range(0, L - 1).long()` and `torch.arange(L).long()` are `torch.equal` for `L = 6` (len 6, values 0..5) — behavior identical, end-inclusive semantics preserved.
- With `warnings.simplefilter("error", UserWarning)`: the current code raises the deprecation `UserWarning`; `torch.arange` constructs cleanly.

### Impact

- No `UserWarning` flooding the training log in DN mode.

### References

- [PyTorch docs: torch.arange](https://pytorch.org/docs/stable/generated/torch.arange.html) — recommended replacement.
- [PyTorch docs: torch.range (deprecated)](https://pytorch.org/docs/stable/generated/torch.range.html) — official deprecation warning ("This function is deprecated and will be removed in a future release because its behavior is inconsistent with Python's range builtin. Instead, use torch.arange(), which produces values in [start, end).").
